connectors/src/connectors/intercom/lib/help_center_permissions.ts

```diff
@@ -0,0 +1,356 @@
+  helpCenterId,
+  withChildren = false,
+}: {
+  connector: Connector;
+  intercomClient: IntercomClient;
+  helpCenterId: string;
+  withChildren?: boolean;
+}): Promise<IntercomHelpCenter | null> {
+  let helpCenter = await IntercomHelpCenter.findOne({
+    where: {
+      connectorId: connector.id,
+      helpCenterId,
+    },
+  });
+
+  if (helpCenter?.permission === "none") {
+    await helpCenter.update({
+      permission: "read",
+    });
+  }
+  if (!helpCenter) {
+    const helpCenterOnIntercom = await fetchIntercomHelpCenter(
+      connector.connectionId,
+      helpCenterId
+    );
+    if (helpCenterOnIntercom) {
+      helpCenter = await IntercomHelpCenter.create({
+        connectorId: connector.id,
+        helpCenterId: helpCenterOnIntercom.id,
+        name: helpCenterOnIntercom.display_name,
+        identifier: helpCenterOnIntercom.identifier,
+        intercomWorkspaceId: helpCenterOnIntercom.workspace_id,
+        permission: "read",
+      });
+    }
+  }
+
+  if (!helpCenter) {
+    logger.error({ helpCenterId }, "[Intercom] Help Center not found.");
+    throw new Error("Help Center not found.");
+  }
+
+  // If withChilren we are allowing the full Help Center.
+  if (withChildren) {
+    const level1Collections = await fetchIntercomCollections(
+      intercomClient,
+      helpCenter.helpCenterId,
+      null
+    );
+    const permissionUpdatePromises = level1Collections.map((c1) =>
+      allowSyncCollection({
+        connector,
+        intercomClient,
+        collectionId: c1.id,
+      })
+    );
+    await Promise.all(permissionUpdatePromises);
+  }
+
+  return helpCenter;
+}
+
+/**
+ * Mark a help center as permission "none" and all children (collections & articles).
+ */
+export async function revokeSyncHelpCenter({
+  connector,
+  helpCenterId,
+}: {
+  connector: Connector;
+  intercomClient: IntercomClient;
+  helpCenterId: string;
+}): Promise<IntercomHelpCenter | null> {
+  const helpCenter = await IntercomHelpCenter.findOne({
+    where: {
+      connectorId: connector.id,
+      helpCenterId,
+    },
+  });
+
+  // Revoke permission for the Help Center
+  if (helpCenter?.permission === "read") {
+    await helpCenter.update({
+      permission: "none",
+    });
+  }
+
+  // Revoke permission for all collections (level 1, level 2 and level 3)
+  await IntercomCollection.update(
+    { permission: "none" },
+    {
+      where: {
+        helpCenterId: helpCenterId,
+      },
+    }
+  );
+
+  // Revoke permission for all articles
+  const level1Collections = await IntercomCollection.findAll({
+    where: {
+      helpCenterId: helpCenterId,
+      parentId: null,
+    },
+  });
+  await Promise.all(
+    level1Collections.map(async (c1) => {
+      await IntercomArticle.update(
+        { permission: "none" },
+        {
+          where: {
+            parents: {
+              [Op.contains]: [c1.collectionId],
+            },
+          },
+        }
+      );
+    })
+  );
+
+  return helpCenter;
```

Comment:
aren't we missing the another level here ?

Reply:
Line 122 we revoke permissions for all Collections of this Help Center (Collections are directly attached to a helpCenterId so easy to grab them all), and for the articles we loop on all level 1 collections to filter on the articles that have one of the level1 collections in their field `parents` which is an array containing the list of parent collections of an article. 

If we were filtering on parentId instead of parents, we should have looped on all 3 level indeed but here we should be good 🤞🏻

Reply:
> which is an array containing the list of parent collections of an article.

Ok thanks, that's the part I was missing 👍  Def makes sense